mlflow/langchain/__init__.py

```diff
@@ -682,6 +684,67 @@ def _convert_ndarray_to_list(data):
             f"for model {self.lc_model.__class__.__name__}"
         )
 
+    def predict_stream(
+        self,
+        data: Any,
+        params: Optional[Dict[str, Any]] = None,
+    ) -> Iterator[str]:
+        """
+        Args:
+            data: Model input data, only single input is allowed.
+            params: Additional parameters to pass to the model for inference.
+
+                .. Note:: Experimental: This parameter may change or be removed in a future
+                    release without warning.
+        Returns:
+            An iterator of model prediction chunks.
+        """
+        from mlflow.langchain.api_request_parallel_processor import process_stream_request
+
+        # TODO: Q: shall we put `callback_handlers` in params ?
+        if isinstance(data, list):
+            raise MlflowException("Langchain model predict_stream only supports single input.")
+
```

Comment:
We need a check here to determine, by an instance method that is set when loading this pyfunc implementation, whether the model was saved with `is_streamable` indication, and only allow for a submission to `process_stream_request` iff the chain type supports streaming. 
This value should be set when logging the model by validating through validating whether the chain elements have the stream-return capabilities within their class definitions. Otherwise, indicated that the model is not streamable and raise an exception if this method is called that explains to the user that streaming support is not available for their logged chain.

Reply:
Make sense.

Reply:
Corresponding checks are added in https://github.com/mlflow/mlflow/pull/11490#discussion_r1548002269 and https://github.com/mlflow/mlflow/pull/11490#discussion_r1548003853